vertx-web-api-contract/src/test/java/io/vertx/ext/web/api/contract/openapi3/OpenAPI3ValidationTest.java

```diff
@@ -779,4 +779,81 @@ public void testQueryOptionalFormExplodeObjectFailure() throws Exception {
 
   }
 
+  /**
+   * Test: query_required_param_form_required_explode_object
+   * Expected parameters sent:
+   * color: R=100&G=200&B=150&alpha=50
+   * Expected response: Validation failure
+   */
+  @Test
+  public void testQueryRequiredParamFormRequiredExplodeObjectFailure() throws Exception {
+    Operation op = testSpec.getPaths().get("/query/form/explode/object").getGet();
+    OpenAPI3RequestValidationHandler validationHandler = new OpenAPI3RequestValidationHandlerImpl(op, op.getParameters(), testSpec, refsCache);
+    loadHandlers("/query/form/explode/object", HttpMethod.GET, false, validationHandler, routingContext -> {
+      RequestParameters params = routingContext.get("parsedParameters");
+
+      RequestParameter colorQueryParam = params.queryParameter("color");
+      assertNotNull(colorQueryParam);
+      assertTrue(colorQueryParam.isObject());
+
+      routingContext.response()
+        .setStatusCode(200)
+        .setStatusMessage("OK")
+        .putHeader("content-type", "application/json")
+        .end(((JsonObject)colorQueryParam.toJson()).encode());
+    });
+
+    String requestURI = "/query/form/explode/object?G=200&B=150&alpha=50";
+
+    testRequest(HttpMethod.GET, requestURI, 400, errorMessage(ValidationException.ErrorType.NOT_FOUND));
+
+  }
+
+  /**
+   * Test: query_required_param_form_optional_explode_object
+   * Expected parameters sent:
+   * color: R=100&G=200&B=150&alpha=50
+   * Expected response: {"color":{"G":"200","B":"150","alpha":"50"}}
+   */
+  @Test
+  public void testQueryRequiredParamFormOptionalExplodeObject() throws Exception {
+    Operation op = testSpec.getPaths().get("/query/form/optional/explode/object").getGet();
+    OpenAPI3RequestValidationHandler validationHandler = new OpenAPI3RequestValidationHandlerImpl(op, op.getParameters(), testSpec, refsCache);
+    loadHandlers("/query/form/optional/explode/object", HttpMethod.GET, false, validationHandler, routingContext -> {
+      RequestParameters params = routingContext.get("parsedParameters");
+
+      RequestParameter colorQueryParam = params.queryParameter("color");
+      assertNotNull(colorQueryParam);
+      assertTrue(colorQueryParam.isObject());
+
+      routingContext.response()
+        .setStatusCode(200)
+        .setStatusMessage("OK")
+        .putHeader("content-type", "application/json")
+        .end(((JsonObject)colorQueryParam.toJson()).encode());
+    });
+
+    String requestURI = "/query/form/optional/explode/object?G=200&B=150&alpha=50";
+
+    testEmptyRequestWithJSONObjectResponse(HttpMethod.GET, requestURI, 200, "OK", new JsonObject("{\"G\":\"200\",\"B\":\"150\",\"alpha\":50}"));
```

Comment:
This test is wrong! The required parameters for `ColorObject` are RGB, so this should return a 400 Bad Request

Reply:
Ya.. I basically turn a 'strict' bug into a 'lenient' bug. Because at least it allows me to query without any parameters. 
I'll see what I can do to properly fix it.

Reply:
What if, as workaround, you just use separated query parameters as optional?

Reply:
Unfortunately we are required to used the exact spec file defined by 3GPP.